src/wrapper/jnienv.rs

```diff
@@ -926,6 +927,56 @@ impl<'a> JNIEnv<'a> {
         Ok(len)
     }
 
+    /// Construct a new array holding objects in class `element_class`.
+    /// All elements are initially set to `initial_element`.
+    pub fn new_object_array(
+        &self,
+        length: jsize,
+        element_class: JClass,
+        initial_element: JObject,
+    ) -> Result<jobjectArray> {
+        non_null!(element_class, "new_object_array element_class argument");
+        Ok(jni_call!(
+            self.internal,
+            NewObjectArray,
+            length,
+            element_class.into_inner(),
+            initial_element.into_inner()
+        ))
+    }
+
+    /// Returns an element of the `jobjectArray` array.
```

Comment:
No, not currently. We can probably go ahead and make the local ref-related functions public, provided we document them. A PR would be welcome :)